src/app/client/src/app/modules/core/components/main-header/main-header.component.ts

```diff
@@ -188,19 +189,56 @@ export class MainHeaderComponent implements OnInit, OnDestroy {
     this.subscription = this.utilService.currentRole.subscribe(async (result) => {
       if (result) {
         this.userType = result;
+        if(this.userType!='adminstrator'){
+          this.setUserPreferences();
+          }
       } else {
         if (this.userService.loggedIn) {
           this.userService.userData$.subscribe((profileData: IUserData) => {
             if (_.get(profileData, 'userProfile.profileUserType.type')) {
               this.userType = profileData.userProfile['profileUserType']['type'];
+              if(this.userType!='adminstrator'){
+                this.setUserPreferences();
+                }
             }
           });
         }
       }
     });
+  }
 
+  getFormConfigs() {
+    this.observationUtilService.browseByCategoryForm()
+      .then((data: any) => {
+        let currentBoard;
+        Array.isArray(this.userPreference.framework.id) ? (currentBoard=this.userPreference.framework.id[0]) : (currentBoard=this.userPreference.framework.id)
```

Comment:
if the object like this.userPreference or the framework is undefined then this will break the entire flow with out checking if we use . operator on the object